documentation/disaster-recovery-testing.md

```diff
@@ -0,0 +1,176 @@
+        - using tba
+- Repo workflows to enable and disable the maintenance page.
+    - see https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md
+    - confirm workflows exists for the selected environment to be tested. Examples;
+        - https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/enable-maintenance.yml
+        - https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/disable-maintenance.yml
+
+## Documentation requirements
+
+Copy the template DR testing document which will be a record of the steps run, time taken, and any issues
+- template doc tba
+
+## Initial set-up
+
+There are no local dependencies to complete the DR test.
+Schedule virtual meeting for the test to take place
+- teams or slack
+- invite relevant technical stakeholders
+Regularly provide updates on the service Slack channel to keep product owners abreast of developments.
+
+## Freeze pipeline
+
+Alert developers that no one should merge to main.
+- In github setings, update the Branch protection rules and set required PR approvers to 6
+
+## Disable statuscake monitoring
+
+If the selcted env has statuscake monitoring, then pause monitoring for the duration of the testing
+
+## Enable maintenance mode
+
+Run the enable maintenance workflow for the service and environment affected
+
+##
+## Follow instructions for the selected scenario
+##
+
+## Scenario 1. Loss of database instance
+
+In case the database instance is lost, the objectives are:
+
+- Recreate the lost postgres database instance
+- Restore data from nightly backup stored in Azure. The point-in-time and snapshot backups created by the Azure Postgres service will not be available if it's been deleted.
+
+### Delete the postgres database instance
+
+Note that you must have a previously created backup on azure storage before starting this step. If not, create one now before continuing.
+
+- Delete the existing postgres database
+    - manually delete via UI https://portal.azure.com/#browse/Microsoft.DBforPostgreSQL%2FflexibleServers
+- confirm it's deleted
+
+If deleting any other resources (scenario 3) then this should be done at this point too.
+
+### Recreate the lost postgres database instance
+
+Run the deploy workflow to recreate the missing postgres database (and any other missing resources).
+
+As the maintenance page has been implemented, you will need to
+- create a branch from main
+- update the env tfvars.json as per https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md#configure-terraform-to-keep-deploying-the-application
+- push the branch to github
+- run the deploy workflow using your branch for the env being tested
+
+If the web app fails to start due to the postgres db missing data
+- update your PR to override the start command to remove any db commands or a dummy command (e.g. sleep 999999)
```

Comment:
Maybe explain the app must be up somehow so the workflow can connect to the DB using konduit

Reply:
changed wording

Reply:
no longer required for the app to be up, so removed wording